test/main-test.js

```diff
@@ -465,6 +465,310 @@ vows.describe('Regexp Tokenizer')
+        assert.deepEqual(t,
+          {
+            type: 0,
+            stack:
+              [
+                { type: types.GROUP, stack: [char('a')], remember: true },
+                char(' '),
+                { type: types.GROUP, stack: [
+                  { type: types.GROUP, stack: [char('b')], remember: true },
+                  char(' '),
+                  { type: types.GROUP, stack: [char('c')], remember: true },
+                ], remember: true },
+                char(' '),
+                char('-'),
+                char(' '),
+                { type: types.REFERENCE, value: 4 },
+              ],
+          });
+      },
+    },
+
+    'Nested capturing groups - \\5 (character)': {
+      topic: ret('(a) ((b) (c)) - \\5'),
+
+      'Reference with more than one digit': t => {
+        assert.deepEqual(t,
+          {
+            type: 0,
+            stack:
+              [
+                { type: types.GROUP, stack: [char('a')], remember: true },
+                char(' '),
+                { type: types.GROUP, stack: [
+                  { type: types.GROUP, stack: [char('b')], remember: true },
+                  char(' '),
+                  { type: types.GROUP, stack: [char('c')], remember: true },
+                ], remember: true },
+                char(' '),
+                char('-'),
+                char(' '),
+                { type: types.CHAR, value: 5 },
+              ],
+          });
+      },
+    },
+
+    // https://github.com/fent/ret.js/pull/39#issuecomment-1008229226
+    'escaped octal numbers': {
+      '\\10': {
+        topic: ret('\\10'),
+        'Tokenizes correctly': t => {
+          assert.deepStrictEqual(t, {
+            type: types.ROOT, stack: [
+              { type: types.CHAR, value: 10 },
+            ],
+          });
+        },
+      },
+      '\\18': {
+        topic: ret('\\18'),
+        'Tokenizes correctly': t => {
+          assert.deepStrictEqual(t, {
+            type: types.ROOT, stack: [
+              { type: types.CHAR, value: 1 },
+              char('8'),
+            ],
+          });
+        },
+      },
+      '\\108': {
+        topic: ret('\\108'),
+        'Tokenizes correctly': t => {
+          assert.deepStrictEqual(t, {
+            type: types.ROOT, stack: [
+              { type: types.CHAR, value: 10 },
+              char('8'),
+            ],
+          });
+        },
+      },
+      '\\107': {
+        topic: ret('\\107'),
+        'Tokenizes correctly': t => {
+          assert.deepStrictEqual(t, {
+            type: types.ROOT, stack: [
+              { type: types.CHAR, value: 107 },
+            ],
+          });
+        },
+      },
+      '\\2': {
+        topic: ret('\\2'),
+        'Tokenizes correctly': t => {
+          assert.deepStrictEqual(t, {
+            type: types.ROOT, stack: [
+              { type: types.CHAR, value: 2 },
+            ],
+          });
+        },
+      },
+      '\\9': {
```

Comment:
i would add an additional test for `'\\90'`, should tokenize as `[char('9'), char('0')`